tests/test_runlib.py

```diff
@@ -447,6 +447,7 @@ def setUpClass(self):
     generate_and_write_rsa_keypair(self.key_path)
     self.key = prompt_import_rsa_key_from_file(self.key_path)
     self.key_pub = prompt_import_rsa_key_from_file(self.key_path + ".pub")
+    self.metadata_directory = './'
```

Comment:
Let's use `tempfile` to create a directory for the tests and write to it. We can then also theoretically play around with its permission bits.